extractor/filesystem/language/golang/gomod/gomod.go

```diff
@@ -140,10 +140,12 @@ func (e Extractor) Extract(ctx context.Context, input *filesystem.ScanInput) ([]
 
 // ToPURL converts an inventory created by this extractor into a PURL.
 func (e Extractor) ToPURL(i *extractor.Inventory) *purl.PackageURL {
+	nameParts := strings.Split(i.Name, "/")
```

Comment:
Since we're doing this in two places can you move the ToPURL logic into a separate library inside language/golang, similar to https://github.com/google/osv-scalibr/blob/main/extractor/filesystem/language/python/internal/pypipurl/pythonpurl.go?